apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -2151,6 +2316,45 @@ fn resolvable_key_applications<'doc>(
     Ok(applications)
 }
 
+fn parse_context(field: &str) -> Result<(String, String), FederationError> {
+    let pattern = Regex::new(
+        r#"^(?:[\n\r\t ,]|#[^\n\r]*)*\$(?:[\n\r\t ,]|#[^\n\r]*)*([A-Za-z_]\w*)([\s\S]*)$"#,
```

Comment:
It is worth noting that the regex from JS was this:
```
/^(?:[\n\r\t ,]|#[^\n\r]*(?![^\n\r]))*\$(?:[\n\r\t ,]|#[^\n\r]*(?![^\n\r]))*([A-Za-z_]\w*(?!\w))([\s\S]*)$/
```
The ported regex is similar but does not include any look-arounds because the `regex` crate does not support them.